SalesforceCommerceCloud/pwa-kit
# @W-14538880@ Fix wishlist bugs

This PR addresses the wishlist issues reported in #1471 and #1489

It also attempts to resolve the issue reported in #1490 but, while it does seem to reduce the frequency, the issue still persists. Further work on that issue will happen in a separate PR.

# Description

Fixes a number of small issues relating to wishlist usability

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Prevent multiple remove from wishlist requests from being sent by the PLP
- Prevent PDP and cart page from creating duplicate wishlist entries if item is already in the wishlist
- Show empty wishlist instead of skeleton once last item has been removed from wishlist

# How to Test-Drive This PR

- Start the app and log in as a registered user (only registered users have wishlists)
- On the PLP - click the heart icon inside a product tile many times in rapid succession
  - Before fix - eventually see errors as you end up sending delete requests for an item that has already been removed from wishlist. Also the heart icon gets stuck in the 'in wishlist' state
  (Note: the issue is still reproducible if you rapidly click the button many times)
  - After fix - no error. heart icon toggles and reflects state of the item in the wishlist
- On PDP - click the add to wishlist button several times
  - Before fix - see that duplicate entries are added to wishlist
  - After fix - only one entry is added. Subsequent clicks display a toast showing item is already in the wishlist
- On cart page - click the add to wishlist button several times
  - Before fix - see that duplicate entries are added to wishlist
  - After fix - only one entry is added. Subsequent clicks display a toast showing item is already in the wishlist
- On account wishlist page - Remove all items from the wishlist
  - Before fix - see that skeleton remains
  - After fix - see an empty wishlist

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
